### PR TITLE
Improved error messages for registerTemplate

### PR DESCRIPTION
**Pull request by didoo**
*Issue #134:*

*Description of changes:*
Added failing `option` values to error messages in `registerTemplate` function. Feel free to update the format/formatting of the messages (if there is a more "standard" way, more consistent with other error messages).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/register/template.js [hunk 1]
```diff
@@ -32,11 +33,11 @@ var fs = require('fs'),
  */
 function registerTemplate(options) {
   if (typeof options.name !== 'string')
-    throw new Error('transform name must be a string');
+    throw new Error('Transform name must be a string: ' + chalk.red(JSON.stringify(options.name)));
```
Comment (dbanksdesign) [inline lib/register/template.js]: One small thing: can you change 'Transform' to 'Template' here? Obviously not your fault, this was a poor copy+paste job by myself a long time ago. 😬

Comment (didoo) [reply lib/register/template.js]: Done.

**Comment by dbanksdesign** [comment]
LGTM! :shipit:

**Comment by chazzmoney** [comment]
Switch to the develop branch for change management purposes

**Pull request merged**